### PR TITLE
Umd task88 fall2025 dgl fraud detection in credit card transactions

### PR DESCRIPTION
**Pull request by hyder1414**
### Summary
Author: Haider Khan
Related to issue #88 

### Phase 1 worked on; 
- **Environment Setup**
  - Structured modular repository with `configs/`, `src/`, `data/`, and `tests/`.
  - Resolved library compatibility issues (`numpy`, `torch`, `torch-geometric`).
- **Data Integration**
  - Joined `train_transaction.csv` and `train_identity.csv` (IEEE-CIS Fraud Detection dataset).
  - Cleaned and stored merged data at `data/processed/merged.parquet`.
- **Feature Engineering**
  - Added transaction-level and account-level features (log amount, time-based, aggregate stats).
  - Output stored as `data/processed/features.parquet`.
- **Graph Construction**
  - Built a bipartite heterogeneous graph (transactions ↔ accounts) using PyTorch Geometric.
  - Saved as `data/artifacts/hetero_graph.pt`.
- **Baseline Model**
  - Logistic Regression (balanced) trained on engineered features.
  - Evaluated precision, recall, F1, and PR-AUC.
  - Metrics logged in `data/artifacts/baseline_metrics.json`.
- **Automation & Testing**
  - Added `Makefile` for reproducible pipeline:
    ```
    make data → make features → make graph → make train_tabular
    ```
  - Implemented basic Pytest checks for graph integrity and split validation.

---

### Dataset
- **Source:** IEEE-CIS Fraud Detection (Kaggle)
- **Files Used:** `train_transaction.csv` and `train_identity.csv`
- **Join Key:** `TransactionID`
- **Label:** `isFraud` (binary)

---

### Next Steps (Phase 2)
- Recompute account-level aggregates using **train-only data** to eliminate label leakage.
- Implement and train **Graph Neural Network (GraphSAGE or GAT)** for transaction-node classification.
- Add **temporal encoding** and explore **edge features**.
- Perform **benchmarking and error analysis** (Precision@K, confusion matrix).
- Integrate **MLOps tracking** (ClearML or W&B) and containerize via Docker.

---

### CODE REVIEW

**Comment by aver81** [comment]
@hyder1414 I'm unable to open your API and example notebooks. also the .md files for API and example are empty. please push your changes before the deadline ends. Remember the API jupyter notebook should only focus on the tool itself, what it is, how to use and some code to show it usage. the API jupyter notebook should not be connected in any way to your project. all project implementation must be contained in the example jupyter notebook

**Comment by hyder1414** [comment]
Hi Aayush. Yes, i saw that and did push yesterday. Thank you. Please let me know if you need anything else.
________________________________
